sycl/plugins/cuda/pi_cuda.hpp

```diff
@@ -453,6 +495,7 @@ struct _pi_queue {
     };
     {
       unsigned int size = static_cast<unsigned int>(compute_streams_.size());
+      std::lock_guard compute_sync_guard(compute_stream_sync_mutex_);
```

Comment:
Acquiring both these locks opens the code up for potential future deadlocks. As such it would be good to add a comment around the declaration of `compute_stream_sync_mutex_` and `compute_stream_mutex_` specifying that the former must be acquired before the latter. Should hopefully help avoid future accidents.

Reply:
done